src/VASSAL/counters/BasicPiece.java

```diff
@@ -378,7 +378,22 @@ public Command setPersistentProperty(Object key, Object val) {
       persistentProps = new HashMap<>();
     }
     final Object oldVal = (val == null) ? persistentProps.remove(key) : persistentProps.put(key, val);
-    return new SetPersistentPropertyCommand (getId(), key, oldVal, val);
+     
+    return setPersistentProperty(key, oldVal, val);
```

Comment:
Have a look at java.lang.Objects.equals(Object a, Object b), this looks like the perfect spot to use it:
`return Objects.equals(oldVal, val) ? null : new SPPC(.....)`

Reply:
Perfect, thanks Yan. I have some other uses for that as well!

Reply:
Code Updated.